util_scripts/Change_Channel_Names.py

```diff
@@ -0,0 +1,262 @@
+            " where w.id in (:ids)"
+        self.plate_query = "select distinct lc.id" \
+            " from Plate as p" \
+            " join p.wells as w" \
+            " join w.wellSamples as ws" \
+            " join ws.image as i" \
+            " join i.pixels as pixels" \
+            " join pixels.channels as c" \
+            " join c.logicalChannel as lc" \
+            " where p.id in (:ids)"
+        self.screen_query = \
+            "select distinct lc.id" \
+            " from Screen as s" \
+            " join s.plateLinks as p_link" \
+            " join p_link.child as p" \
+            " join p.wells as w" \
+            " join w.wellSamples as ws" \
+            " join ws.image as i" \
+            " join i.pixels as pixels" \
+            " join pixels.channels as c" \
+            " join c.logicalChannel as lc" \
+            " where s.id in (:ids)"
+        self.dataset_query = \
+            "select distinct lc.id" \
+            " from Dataset as d" \
+            " join d.imageLinks as dil" \
+            " join dil.child as i" \
+            " join i.pixels as pixels" \
+            " join pixels.channels as c" \
+            " join c.logicalChannel as lc" \
+            " where d.id in (:ids)"
+        self.project_query = \
+            "select distinct lc.id" \
+            " from Project p" \
+            " join p.datasetLinks as links" \
+            " join links.child as dataset" \
+            " join dataset.imageLinks as dil" \
+            " join dil.child as i" \
+            " join i.pixels as pixels" \
+            " join pixels.channels as c" \
+            " join c.logicalChannel as lc" \
+            " where p.id in (:ids)"
+        self.image_query = \
+            "select distinct lc.id" \
+            " from Image as i" \
+            " join i.pixels as pixels" \
+            " join pixels.channels as c" \
+            " join c.logicalChannel as lc" \
+            " where i.id in (:ids)"
+
+    def getLcIdsList(self, query):
+        params = omero.sys.ParametersI()
+        params.addIds(self.ids)
+        lc_ids = self.query_service.projection(query, params)
+        if len(lc_ids) == 0:
+            return None
+        lc_ids = [lc_id[0] for lc_id in lc_ids]
+        return [lc_id.getValue() for lc_id in lc_ids]
+
+    def renameBatch(self, lc_ids, step):
+        lc_ids_batch = random.sample(lc_ids, step)
+        params = omero.sys.ParametersI()
+        params.addIds(lc_ids_batch)
+        number_of_channels = len(self.new_channel_names)
+        paging = self.image_paging
+        counter = 0
+        exit = False
+        while len(lc_ids_batch) > 0:
+            params.page(counter * paging, paging)
+            counter += 1
+            image_list = self.query_service.findAllByQuery(
+                self.get_image_query, params)
+            print "Retrived %i images" % len(image_list)
+            for image in image_list:
+                check_lc = image.getPixels(0).getChannel(0). \
```

Comment:
You can use the convenience method `getPrimaryPixels()` in place of `getPixels(0)`.